pkg/controller/route53resolver/resolverendpoint/hooks.go

```diff
@@ -87,3 +90,22 @@ func preUpdate(_ context.Context, cr *v1alpha1.ResolverEndpoint, obj *svcsdk.Upd
 	obj.ResolverEndpointId = aws.String(meta.GetExternalName(cr))
 	return nil
 }
+
+func postObserve(_ context.Context, cr *svcapitypes.ResolverEndpoint, obj *svcsdk.GetResolverEndpointOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	switch aws.StringValue(obj.ResolverEndpoint.Status) {
+	case string(svcapitypes.ResolverEndpointStatus_SDK_OPERATIONAL):
+		cr.SetConditions(xpv1.Available())
+	case string(svcapitypes.ResolverEndpointStatus_SDK_CREATING):
+		cr.SetConditions(xpv1.Creating())
+	case string(svcapitypes.ResolverEndpointStatus_SDK_UPDATING), string(svcapitypes.ResolverEndpointStatus_SDK_ACTION_NEEDED):
```

Comment:
Are we sure the resource is not available during update or action needed?

Reply:
UPDATING: Resolver is associating or disassociating one or more network interfaces with this endpoint / yes If all addresses are updating 

ACTION_NEEDED: This endpoint is unhealthy, and Resolver can't automatically recover it. / yes